src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/LanguageServices/SyntaxFactsService/ISyntaxFactsService.cs

```diff
@@ -2,405 +2,15 @@
-        /// Returns true if the given character is a character which may be included in an
-        /// identifier to specify the type of a variable.
-        /// </summary>
-        bool IsTypeCharacter(char c);
-
-        bool IsBindableToken(SyntaxToken token);
-
-        bool IsInStaticContext(SyntaxNode node);
-        bool IsUnsafeContext(SyntaxNode node);
-
-        bool IsInNamespaceOrTypeContext(SyntaxNode node);
-
-        bool IsBaseTypeList(SyntaxNode node);
-
-        bool IsAnonymousFunction(SyntaxNode n);
-
-        bool IsInConstantContext(SyntaxNode node);
-        bool IsInConstructor(SyntaxNode node);
-        bool IsMethodLevelMember(SyntaxNode node);
-        bool IsTopLevelNodeWithMembers(SyntaxNode node);
-        bool HasIncompleteParentMember(SyntaxNode node);
-
-        /// <summary>
-        /// A block that has no semantics other than introducing a new scope. That is only C# BlockSyntax.
-        /// </summary>
-        bool IsScopeBlock(SyntaxNode node);
-
-        /// <summary>
-        /// A node that contains a list of statements. In C#, this is BlockSyntax and SwitchSectionSyntax.
-        /// In VB, this includes all block statements such as a MultiLineIfBlockSyntax.
-        /// </summary>
-        bool IsExecutableBlock(SyntaxNode node);
-        SyntaxList<SyntaxNode> GetExecutableBlockStatements(SyntaxNode node);
-        SyntaxNode FindInnermostCommonExecutableBlock(IEnumerable<SyntaxNode> nodes);
-
-        /// <summary>
-        /// A node that can host a list of statements or a single statement. In addition to
-        /// every "executable block", this also includes C# embedded statement owners.
-        /// </summary>
-        bool IsStatementContainer(SyntaxNode node);
-        IReadOnlyList<SyntaxNode> GetStatementContainerStatements(SyntaxNode node);
-
-        bool AreEquivalent(SyntaxToken token1, SyntaxToken token2);
-        bool AreEquivalent(SyntaxNode node1, SyntaxNode node2);
-
-        string GetDisplayName(SyntaxNode node, DisplayNameOptions options, string rootNamespace = null);
-
-        SyntaxNode GetContainingTypeDeclaration(SyntaxNode root, int position);
-        SyntaxNode GetContainingMemberDeclaration(SyntaxNode root, int position, bool useFullSpan = true);
-        SyntaxNode GetContainingVariableDeclaratorOfFieldDeclaration(SyntaxNode node);
-
-        SyntaxToken FindTokenOnLeftOfPosition(SyntaxNode node, int position, bool includeSkipped = true, bool includeDirectives = false, bool includeDocumentationComments = false);
-        SyntaxToken FindTokenOnRightOfPosition(SyntaxNode node, int position, bool includeSkipped = true, bool includeDirectives = false, bool includeDocumentationComments = false);
-
-        void GetPartsOfParenthesizedExpression(SyntaxNode node, out SyntaxToken openParen, out SyntaxNode expression, out SyntaxToken closeParen);
         SyntaxNode Parenthesize(SyntaxNode expression, bool includeElasticTrivia = true, bool addSimplifierAnnotation = true);
```

Comment:
The 3 methods left behind in this interface definition are not pure syntactic data query methods. They should arguably be moved to a separate service/provider. These access Workspace layer annotations, so I have kept them in this type itself to avoid cloning annotations.